src/vic.h

```diff
@@ -74,12 +75,13 @@
 #define VIC_ERR_INVALID_ARGS 3
 
 /* vic-serial.c */
-extern char vic_buffer[VIC_BUFFER_SIZE + 1];
-extern uint8_t vic_buffer_len;
+extern uint8_t vic_buffer_in_len;
+extern char vic_buffer_in[VIC_BUFFER_IN_SIZE + 1];
+extern char vic_buffer_out[VIC_BUFFER_OUT_SIZE + 1];
 
-#define vic_printf(format, ...) do { char _vic_buff[VIC_BUFFER_SIZE];\
-    snprintf(_vic_buff, VIC_BUFFER_SIZE, (format), __VA_ARGS__);\
-    vic_print(_vic_buff); } while(0)
+#define vic_printf(format, ...) do {\
+    snprintf(vic_buffer_out, VIC_BUFFER_OUT_SIZE + 1, (format), __VA_ARGS__);\
```

Comment:
@mrshu vic_buffer_out is always used only for a moment and then is printed, so that doesn't make problem